src/uploader.rs

```diff
@@ -86,6 +93,8 @@ async fn upload_layer(
                 bytes_written_tx.send(bytes_written).unwrap();
             })),
         );
+        
+        //@ex encoder calculates hash for compressed content in streaming fashion
```

Comment:
// NOTE(ex)
// TODO(ex) 

// generic comments that helps anyone, you don't need to attribute it